spinalcordtoolbox/utils/shell.py

```diff
@@ -329,6 +329,7 @@ def _split_lines(self, text, width):
         offsets = [re.match("^[ \t]*", line).group(0) for line in lines]
         wrapped = []
         for i, li in enumerate(lines):
+            li = li.rstrip()  # strip trailing whitespace
```

Comment:
I was curious why `rstrip` rather than `lstrip` or even `strip`, but I see that this matches the behaviour of `textwrap.fill`: it preserves leading whitespace, but discards trailing whitespace. I guess leading whitespace is likely to be intentional formatting.

Reply:
That was exactly my thinking! Leading whitespace is frequently used when indenting lists, and in fact our formatter explicitly picks up on this:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/c515397ec53616cbe3b834be24f53016537cccec/spinalcordtoolbox/utils/shell.py#L341-L342